src/main/java/net/datafaker/transformations/JavaObjectTransformer.java

```diff
@@ -167,11 +171,12 @@ private Object getObject(Schema<Object, ?> schema, Object result, Constructor<?>
     }
 
     private boolean hasParameterlessPublicConstructor(Class<?> clazz) {
-        for (Constructor<?> constructor : clazz.getConstructors()) {
-            if (constructor.getParameterCount() == 0) {
-                return true;
-            }
-        }
-        return false;
+        return getParameterlessPublicConstructor(clazz).isPresent();
+    }
+
+    private Optional<Constructor<?>> getParameterlessPublicConstructor(Class<?> clazz) {
+        return Arrays.stream(clazz.getConstructors())
```

Comment:
Does clazz.getConstructors() restrict to public? If not then there's a condition missing to conform to the function name

Reply:
1. `clazz.getDeclaredConstructors()` returns ALL constructors
2. `clazz.getConstructors()` returns only PUBLIC constructors.